services/src/datasets/create_from_workflow.rs

```diff
@@ -186,6 +187,20 @@ pub async fn schedule_raster_dataset_from_workflow_task<C: SessionContext>(
     info: RasterDatasetFromWorkflow,
     compression_num_threads: GdalCompressionNumThreads,
 ) -> error::Result<TaskId> {
+    if let Some(dataset_name) = &info.name {
+        let db = ctx.db();
+
+        let potential_id = db.resolve_dataset_name_to_id(dataset_name).await;
+
+        if let Ok(id) = potential_id {
```

Comment:
couldn't the dataset name already exists even though we get an Error back? E.g. the error could be `InvalidDatasetIdNamespace` or `InvalidNamespace` or even an Authorization error?

Reply:
the resolve method now returns an option. This allows to handle more cases.

https://github.com/geo-engine/geoengine/pull/902/commits/9e7daef753cd65886efc164e9f7edda11eb743c7